quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/QuarkusPropertiesDistTest.java

```diff
@@ -132,6 +133,26 @@ void testUnknownQuarkusBuildTimePropertyApplied(LaunchResult result) {
                 .body(containsString("jvm_gc_"));
     }
 
+    @Test
+    @Launch({ "start", "--http-enabled=true", "--hostname-strict=false", "--config-keystore=keystore" })
+    @Order(10)
+    void testMissingSmallRyeKeyStoreProperty(LaunchResult result) {
+        CLIResult cliResult = (CLIResult) result;
+        cliResult.assertError("The config property smallrye.config.source.keystore.kc-default.password is required but it could not be found in any config source");
+        cliResult.assertNoBuild();
+    }
+
+    @Test
+    @Order(11)
+    void testSmallRyeKeyStoreConfigSource(KeycloakDistribution distribution) {
```

Comment:
Is it actually covering loading the database credentials from the Keystore?

Reply:
Yes, the log level as it is easiest to test.